Mathlib/Combinatorics/Digraph/Basic.lean

```diff
@@ -0,0 +1,236 @@
+/-
+Copyright (c) 2024 Kyle Miller, Jack Cheverton. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kyle Miller, Jack Cheverton, Jeremy Tan
+-/
+import Mathlib.Order.CompleteBooleanAlgebra
+import Mathlib.Data.Fintype.Pi
+
+/-!
+# Digraphs
+
+This module defines directed graphs on a vertex type `V`. Although the structure is definitionally
+equivalent to a `Quiver` in a specialized universe, the intention is to consider graph theory
+language as basic. Furthermore, `Digraph` is not intended to be used as a class.
+
+## Main definitions
+
+* `Digraph` is a structure for relations. Unlike `SimpleGraph`, the relation does not need to be
+  symmetric or irreflexive.
+
+* `CompleteAtomicBooleanAlgebra` instance: Under the subgraph relation, `Digraph` forms a
+  `CompleteAtomicBooleanAlgebra`. In other words, this is the complete lattice of spanning subgraphs
+  of the complete graph.
+-/
+
+open Finset Function
+
+/-- A digraph is a relation `Adj` on a vertex type `V`.
+The relation describes which pairs of vertices are adjacent. -/
+@[ext]
+structure Digraph (V : Type*) where
+  /-- The adjacency relation of a digraph. -/
+  Adj : V → V → Prop
+
+noncomputable instance {V : Type*} [Fintype V] : Fintype (Digraph V) := by
+  classical exact Fintype.ofInjective Digraph.Adj (fun _ _ ↦ Digraph.ext)
```

Comment:
Can't you use `Fintype.ofSurjective` here to get this computabily?

Reply:
Move this after `mk'` and then you can use it with `Fintype.ofBijective` (it saves needing that `Digraph V` have `DecidableEq`):
```lean
instance {V : Type*} [DecidableEq V] [Fintype V] : Fintype (Digraph V) :=
  Fintype.ofBijective Digraph.mk' <| by
    refine ⟨Embedding.injective _, ?_⟩
    classical
    intro G
    use fun v w ↦ G.Adj v w
    ext v w
    simp
```